client/lib/domains/dns/index.js

```diff
@@ -86,32 +75,44 @@ function isValidData( data, type ) {
 			return data.match( /^[a-f0-9\:]+$/i );
 		case 'CNAME':
 		case 'MX':
-			return isValidDomainName( data, type );
+			return isValidDomain( data );
 		case 'TXT':
 			return data.length < 256;
 	}
 }
 
-function getNormalizedData( fieldValues, selectedDomainName ) {
-	var data = fieldValues;
-
-	data.data = getFieldWithDot( data.data );
-
-	if ( includes( [ 'A', 'AAAA' ], data.type ) ) {
-		data.name = removeTrailingDomain( data.name, selectedDomainName );
+function getNormalizedData( record, selectedDomainName ) {
+	const normalizedRecord = Object.assign( {}, record );
+	normalizedRecord.data = getFieldWithDot( record.data );
+	normalizedRecord.name = getNormalizedName( record.name, record.type, selectedDomainName );
+	if ( record.target ) {
+		normalizedRecord.target = getFieldWithDot( record.target );
 	}
 
-	data.name = getFieldWithDot( data.name );
+	return normalizedRecord;
+}
+
+function getNormalizedName( name, type, selectedDomainName ) {
```

Comment:
Can you add some docs about what these functions do, or rename the functions to describe what they're doing?